### DISCUSSION TITLE
Permit data-foo attribute to be keep

### DISCUSSION THREAD

**Discussion by khellendros74** [orchidjs/tom-select#325] [Q&A]
Creating a Option element with extra attribute 

```
<select id="mySelect">
<option data-foo="otherdata1" value="1">One</option>
</select>
...
new TomSelect("#mySelect", {});

let option = new Option("Two", 2);
option.setAttribute("data-foo", "otherdata2");
select.tomselect.addOption(option);

option = new Option("Three", 3);
option.setAttribute("data-foo", "otherdata3");
select.option.add(option);
```
now what ever way it was created, selected option is missing data-foo

```
let selectedOption = document.getElementById("mySelect").querySelector('option:checked');
selectedOption.getAttribute("data-context");

```

i found something about "dataAttr" in the documentation but did not found a way to use it

**Comment by khellendros74** [comment]
somewhere lost in documentation
https://tom-select.js.org/examples/options/

**Reply by ghost** [reply]
It doesn't answer at all the question and it's not written in the mentioned link or no longer is.

Edit: https://github.com/orchidjs/tom-select/discussions/444

**Comment by jaguilarsn** [comment]
Hola tuve el mismo problema y con la función escape consegui lo deseado dejo el ejemplo:
<select id="horario" name="horario" placeholder="¿Cómo te gustaría estudiar?" class="form-control">
<p>
let opciones = [
    { value: '1', text: 'a',  customProp: 'D'},
    { value: '2', text: 'b',  customProp: 'O'},
    { value: '3', text: 'c',  customProp: 'X'},
    { value: '4', text: 'd',  customProp: 'F'},
    { value: '5', text: 'e',  customProp: 'S'}
  ]
</p>
<p>
  let selectHoraio = new TomSelect('#horario',{
    options: opciones,
    render: {
      option: function(data, escape) {
        return `<div customProp='${escape(data.customProp)}'> 
        ${escape(data.text)}
        </div>`;
      },
      item: function(data, escape) {
        return `<div title="${escape(data.value)}">${escape(data.text)}</div>`;
      }
    }
  });
  
    selectHoraio.on('change', function () {
      var nuevoValor = selectHoraio.getValue().toUpperCase().trim();
      var valorPropiedad = selectHoraio.getOption(selectHoraio.getValue()).getAttribute('customProp');
      console.log(valorPropiedad)
      console.log('Nuevo valor seleccionado:', nuevoValor);
    });
</p>

**Comment by ve3** [comment]
The answer about set `data-` attributes to the `<option>` is at  https://github.com/orchidjs/tom-select/discussions/444
